File: src/main/java/com/example/probation/validatiion/annotation/UniqueEmail.java
package com.example.probation.validatiion.annotation;

import com.example.probation.validatiion.validator.EmailValidator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Target({ElementType.METHOD, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Constraint(validatedBy = EmailValidator.class)
public @interface UniqueEmail {
    String message() default "{email.not.unique}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/com/example/probation/validation/annotation/UniqueEmail.kt
package com.example.probation.validation.annotation

import com.example.probation.validation.validator.EmailValidator
import com.example.probation.validation.validator.NewEmailValidator
import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.FIELD, AnnotationTarget.CLASS)
@Constraint(validatedBy = [EmailValidator::class, NewEmailValidator::class])
annotation class UniqueEmail(
    val message: String = "{email.not.unique}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
